### ISSUE TITLE
[JupyterHub] The JH deploymentconfig rolls out new deployments very frequently

### ISSUE THREAD

**Issue by 4n4nd** [opendatahub-io/odh-manifests#358]
**Describe the bug**
The Jupyterhub deploymentconfig rolls out new deployments very often with the following status:
```yaml
status:
  observedGeneration: 26
  details:
    message: image change
    causes:
      - type: ImageChange
        imageTrigger:
          from:
            kind: DockerImage
            name: 'jupyterhub-img:latest'
```
![Screenshot from 2021-03-23 12-05-19](https://user-images.githubusercontent.com/22333506/112178451-1f542a00-8bd0-11eb-9485-7b64b8548598.png)

**Steps To Reproduce**
1. Deploy Jupyterhub using ODH v1.0.5

**Expected behavior**
New Rollouts should not be triggered automatically

**Open Data Hub Version**
[kfdef link](https://github.com/operate-first/apps/blob/946e8d021fbc23ef413d596987d1fc664a9ceb58/odh/base/jupyterhub/kfdef.yaml)

**OpenShift Version**
Version #: v4.7.0
Provider: Baremetal

**Comment by LaVLaS** [comment]
Investigating the same cluster that experienced this issue, it appears that some event triggered this issue on a 10hr cycle over 2 days that were still available in the logs. I can't reproduce this on another cluster using the same kfdef you uploaded. 

I will continue to track this issue since introducing an endless deployment loop was a concern when we implemented the `imageChange` trigger in #334.

**Comment by nakfour** [comment]
We added a trigger for image changes https://github.com/opendatahub-io/odh-manifests/pull/360
